core/dataset.cc

```diff
@@ -968,6 +973,11 @@ generateRowsWhere(const SqlBindingScope & scope,
 
                             for (auto& c : inExpression->tuple->clauses) {
                                 ExpressionValue v = c->constantValue();
+
+                                //casting other types to string will give a different result than non-optimized path.
+                                if (!v.isString())
```

Comment:
This means that row names are always interpreted as string and they will never match a IN expression that is not a string, correct?